### PR TITLE
Testcases for issue #12

### PR DESCRIPTION
**Pull request by bastbnl**
Failing testcases

### CODE REVIEW

File: tests/templates/test_issue#12_intermediate.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% extends 'test_issue#12_base.txt' %}
+
+{% block block1 %}
+{{ block.super }}
```
Comment (clokep) [inline tests/templates/test_issue#12_intermediate.html]: Isn't this not valid in Django templating engine? Are you using Jinja2?

Comment (bastbnl) [reply tests/templates/test_issue#12_intermediate.html]: @clokep  I'm using Django. Are you referring to the {{block.super}}? Or to the additional block I created afterwards? 

Well, I can see the block names (in context), so I'm guessing it's valid. I guess I can try the same in a webpage and then have it rendered by Django

Comment (clokep) [reply tests/templates/test_issue#12_intermediate.html]: I'm referring the `{{ block.super }}` call.

Comment (clokep) [reply tests/templates/test_issue#12_intermediate.html]: You're right though, I'm just mixing up the syntax between the two. `block.super` is a Django templating engine thing: https://docs.djangoproject.com/en/1.11/ref/templates/language/

**Comment by clokep** [comment]
OK, so your `test_issue12_empty_block` seems to be passing, did you expect that to fail?

The `test_issue12_subblock` looks correct, but is failing so yeah...definitely looks like a bug. 😢

**Comment by bastbnl** [comment]
Well, I expected the first test to succeed.

I suspect the issue I'm seeing is caused by `_build_block_context`, but I don't really get why, since a node by that name is in `context.render_context[BLOCK_CONTEXT_KEY].blocks`. The fix solved that issue but raised another.

**Comment by clokep** [comment]
I'm unsure if this is the case of your issues or not, but the test-case is a bit confusing. Sometimes `block_1` is used and sometimes `block1`, so I think that needs to be fixed first!

**Comment by clokep** [comment]
OK, just restarted the build and the errors look saner now? It seems like some content is being duplicated, which isn't good, and that one block can't be found?

**Comment by bastbnl** [comment]
Exactly!

**Comment by clokep** [comment]
Well, definitely seems like a bug! I don't know off the top of my head what's causing it though, so I'll need to dig a bit. Did you have any luck?

**Comment by bastbnl** [comment]
Well, I did find something that looks like a workaround but it crashed on me as soon as I started it. I don't have too much time on my hands and I'll try to look into it tomorrow.

**Comment by clokep** [comment]
Closing due to lack of progress. Please reopen if you intend to continue working on this.

**Pull request closed without merging**